src/csm_testing/tests/iuf_cleanup/__main__.py

```diff
@@ -0,0 +1,220 @@
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+This script executes deletion of logs , workflows and directories created for an iuf activity.
+"""
+
+import subprocess
+import sys
+from pathlib import Path
+from csm_testing.lib.iuf_constants import MEDIA_DIR
+
+LOG_DIR = "/etc/cray/upgrade/csm/iuf"
+
+
+def get_workflows(activity_name):
+    """
+    Function to get the list of workflows for the activity.
+    Args:
+        activity_name(str): The activity name whose workflows list needs to be generated
+    Returns:
+        configmaps(List[str]): List of workflows for the activity
+    """
+    command = f"kubectl get workflow -n argo -o \
+custom-columns=NAME:.metadata.name|grep {activity_name}"
+    workflows = []
+    try:
+        result = subprocess.run(
+            command,
+            shell=True,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+        )
+        if not result.returncode:
+            workflows = result.stdout.splitlines()
+    except subprocess.CalledProcessError as err:
+        return err.returncode
+    return workflows
+
+
+def get_configmaps(activity_name):
+    """
+    Function to get the list of configmaps for the activity.
+    Args:
+        activity_name(str): The activity name whose configmaps list needs to be generated
+    Returns:
+        configmaps(List[str]): List of configmaps for the activity
+    """
+    command = f"kubectl get configmap -n argo -o \
+custom-columns=NAME:.metadata.name|grep {activity_name}"
+    configmaps = []
+    try:
+        result = subprocess.run(
+            command,
+            shell=True,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+        )
+        if not result.returncode:
+            configmaps = result.stdout.splitlines()
+    except subprocess.CalledProcessError as err:
+        return err.returncode
+    return configmaps
+
+# pylint: disable=too-many-branches
```

Comment:
If pylint is raising this error, it is a strong indication that the function in question should be refactored, possibly splitting it into multiple functions. This is one of the more important pylint errors to actually address, because it can make code much harder to understand and maintain.